### DISCUSSION TITLE
Interpreting the Tensorboard

### DISCUSSION THREAD

**Discussion by Thresher12** [coqui-ai/TTS#2318] [General Q&A]
Sorry, I'm a bit new at this. I'm training on a custom dataset and I'm not sure how to proceed since I'm completely lost when trying to understand the Tensorboard. 

Can someone give me a cliff notes or point me on to an easy to understand resource on what these different graphs mean? At least the generic ones that don't specifically relate only to capacitron. I know that the losses and alignment errors are supposed to go  down with training and thats basically it. Theres a dearth of straightforward information when I try to google the different terms like 'avg decoder loss' and 'the difference between Epoch and Evaluation stats'. Much of the info out there is just how to set things up assuming you already know how to interpret everything. 

Config code

```
dataset_config = BaseDatasetConfig(
    name="ljspeech",
    meta_file_train="metadata.csv",
    path=os.path.join(output_path, "data/"),
)

audio_config = BaseAudioConfig(
    sample_rate=22050,
    do_trim_silence=True,
    trim_db=60.0,
    signal_norm=False,
    mel_fmin=0.0,
    mel_fmax=11025,
    spec_gain=1.0,
    log_func="np.log",
    ref_level_db=20,
    preemphasis=0.0,
)

# Using the standard Capacitron config
capacitron_config = CapacitronVAEConfig(capacitron_VAE_loss_alpha=1.0, capacitron_capacity=50)

config = Tacotron2Config(
    run_name="Capacitron-Tacotron2",
    audio=audio_config,
    capacitron_vae=capacitron_config,
    use_capacitron_vae=True,
    batch_size=32,  # Tune this to your gpu
    max_audio_len=8 * 22050,  # Tune this to your gpu
    min_audio_len=1 * 22050,
    eval_batch_size=16,
    num_loader_workers=8,
    num_eval_loader_workers=8,
    precompute_num_workers=24,
    run_eval=True,
    test_delay_epochs=25,
    ga_alpha=0.0,
    r=2,
    optimizer="CapacitronOptimizer",
    optimizer_params={"RAdam": {"betas": [0.9, 0.998], "weight_decay": 1e-6}, "SGD": {"lr": 1e-5, "momentum": 0.9}},
    attention_type="dynamic_convolution",
    grad_clip=0.0,  # Important! We overwrite the standard grad_clip with capacitron_grad_clip
    double_decoder_consistency=False,
    epochs=1000,
    text_cleaner="phoneme_cleaners",
    use_phonemes=True,
    phoneme_language="en-us",
    phonemizer="espeak",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    stopnet_pos_weight=15,
    print_step=25,
    print_eval=True,
    mixed_precision=False,
    sort_by_audio_len=True,
    seq_len_norm=True,
    output_path=output_path,
    datasets=[dataset_config],
    lr=1e-3,
    lr_scheduler="StepwiseGradualLR",
    lr_scheduler_params={
        "gradual_learning_rates": [
            [0, 1e-3],
            [2e4, 5e-4],
            [4e5, 3e-4],
            [6e4, 1e-4],
            [8e4, 5e-5],
        ]
    },
    scheduler_after_epoch=False,  # scheduler doesn't work without this flag
    # Need to experiment with these below for capacitron
    loss_masking=False,
    decoder_loss_alpha=1.0,
    postnet_loss_alpha=1.0,
    postnet_diff_spec_alpha=0.0,
    decoder_diff_spec_alpha=0.0,
    decoder_ssim_alpha=0.0,
    postnet_ssim_alpha=0.0,
)

```

![trainingstats1](https://user-images.githubusercontent.com/10659223/216032129-9d0078ee-6302-461f-ba7b-09ede56ca382.png)

![trainingstats2](https://user-images.githubusercontent.com/10659223/216032173-3cd705f4-d46c-4930-a17c-f16d53126529.png)

![trainingstats3](https://user-images.githubusercontent.com/10659223/216032267-69ef9438-ccac-4e43-bf3e-a2e71533ea24.png)

![trainingstats4](https://user-images.githubusercontent.com/10659223/216032292-6178eb94-f3c3-4a53-8b2a-882c8ea2d58c.png)

![trainingstats5](https://user-images.githubusercontent.com/10659223/216032324-4e35df7e-61e0-4a8e-bbbf-f29511e11bd4.png)

**Comment by p0p4k** [comment]
Hello, did you train a model successfully? The main aim is to focus on attention alignment going diagonal and seeing if your spectrogram resembles something to a real spectrogram.
